content/common/cli-reference.md

```diff
@@ -1068,7 +1100,51 @@ TRIGGER OPTIONS:
 
 {{ blockEnd }}
 
-> **Please note:** If the `-f` or `--from` options do not accurately infer the intended registry or `.toml` file for your application, then you can explicitly specify either the `--from-registry` or  `--from-file` options to clarify this.
+> **Please note:** If the `-f` or `--from` options do not accurately infer the intended registry or `.toml` file for your application, then you can explicitly specify either the `--from-registry` or `--from-file` options to clarify this.
+
+### Watch
+
+In order to configure the files that are watched you need to configure the `component.build.watch` parameter in your spin.toml.
+
+```toml
+[component.build]
+# Example watch configuration for a Rust application
+watch = ["src/**/*.rs", "spin.toml", "Cargo.toml"]
+```
+
+{{ tabs "spin-version" }}
+
+{{ startTab "v1.1.0"}}
+
+<!-- @selectiveCpy -->
+
+```console
+$ spin watch --help
+
+spin-watch
+Rebuild and restart the Spin application when files changes
+
+USAGE:
+    spin watch [OPTIONS] [UP_ARGS]...
+
+ARGS:
+    <UP_ARGS>...    Arguments to be passed through to spin up
+
+OPTIONS:
+    -c, --clear                       Clear the screen before each run
+    -d, --debounce <DEBOUNCE>         Set the timeout between detected change and re-execution, in
+                                      milliseconds [default: 100]
+    -f, --file <APP_MANIFEST_FILE>    Path to spin.toml [default: spin.toml]
+    -h, --help                        Print help information
+        --skip-build                  Skip running spin build and only run spin up
+        --watch-assets                Only watch component source and files ignoring watch
```

Comment:
Might be good to wordsmith this name if it actually watches the Wasm module as well as the assets (again, sorry for missing this when the PR was out)

Reply:
Resolved by our earlier conversations.